File: scripts/test_task_planner.py
import requests
import json
import sys

def test_task_planner():
    """Test the Cluely task planner functionality directly."""
    url = "http://localhost:8888/command"
    
    # Use a command that should trigger the task planner
    command = "create a text file on my Desktop called test_task_planner.txt with content 'This is a test of the task planner system'"
    
    print(f"Sending command: {command}")
    
    # Make the request
    response = requests.post(
        url,
        json={
            "command": command,
            "context": []
        }
    )
    
    # Parse response
    data = response.json()
    print("\nResponse:")
    print(json.dumps(data, indent=2))
    
    # Check if task planner was used
    method = data.get("metadata", {}).get("method", "unknown")
    print(f"\nMethod used: {method}")
    print(f"Is AI Response: {data.get('is_ai_response', False)}")
    
    # Check if actions were performed
    actions = data.get("metadata", {}).get("actions_performed", [])
    if actions:
        print("\nActions performed:")
        for i, action in enumerate(actions):
            print(f"{i+1}. Type: {action.get('action_type', 'unknown')}")
            if action.get("output"):
                print(f"   Output: {action.get('output')}")
            if action.get("error"):
                print(f"   Error: {action.get('error')}")

if __name__ == "__main__":
    test_task_planner()

File: scripts/test-task-planner.js
const axios = require('axios');

// This is a simple test script to test the task planner functionality
// It sends a command to the backend API and prints the response

const backendPort = 8889; // Using the alternate port we started on
const backendUrl = `http://localhost:${backendPort}`;

// Test a complex command that should trigger the task planner
async function testTaskPlanner() {
  try {
    // Using a more complex multi-step command
    const command = "find all text files in my Desktop folder and create a new file called summary.txt with their names and sizes";
    console.log(`Testing task planner with command: "${command}"`);
    
    const response = await axios.post(`${backendUrl}/command`, {
      command,
      context: []
    });
    
    console.log("Response:");
    console.log(JSON.stringify(response.data, null, 2));
    
    // Check if the task planner was used
    const metadata = response.data.metadata || {};
    console.log(`Method used: ${metadata.method || 'unknown'}`);
    console.log(`Is AI response: ${response.data.is_ai_response ? 'Yes' : 'No'}`);
    
    // If actions were performed, list them
    if (metadata.actions_performed && metadata.actions_performed.length > 0) {
      console.log("\nActions performed:");
      metadata.actions_performed.forEach((action, i) => {
        console.log(`${i + 1}. Type: ${action.action_type || 'unknown'}`);
        if (action.output) console.log(`   Output: ${action.output}`);
        if (action.error) console.log(`   Error: ${action.error}`);
      });
    }
    
  } catch (error) {
    console.error("Error testing task planner:", error.message);
    if (error.response) {
      console.error("Response data:", error.response.data);
      console.error("Response status:", error.response.status);
    }
  }
}

// Run the test
testTaskPlanner();
